README.md

```diff
@@ -16,6 +16,92 @@ See [PROPOSAL.md](PROPOSAL.md) for the proposal for this project.
+
+All of the above (demo app, Concourse and the operator) run on a kubernetes cluster. All images are from (or through) Docker Hub.
+
+## Demo App
+
+This is just a demo app built with Angular. It displays some text. It has tests.
+
+The [demo-app-statefulset.yaml](demo-app-statefulset.yaml) configuration defines a stateful set with three pods and a service with type NodePort.
+
+We use a stateful set instead of a replica set because of the nice predictable and readable names (e.g. demo-app-0, demo-app-1 and demo-app-2) and the service type NodePort because the cluster we deploy on has an Nginx reverse proxy.
+
+## Concourse
+
+Concourse is a CI/CD system which allows creating pipelines consisting of jobs, where each job runs in its own container. We use Concourse mainly because it is a cool project and does what we want: running jobs (which includes tests, of course).
+
+### Deployment Customization
+
+The customization for the Concourse deployment is found in [kustomization.yaml](kustomization.yaml), or rather: the starting point is found in that file. The configuration is split into manageable parts, and therefore this file just contains the delegation to the folders postgresql and concourse, which contain the actual customization for those parts.
+
+```yaml
+# kustomization.yaml
+resources:
+  - postgresql
+  - concourse
+```
+
+This pattern of splitting the configuration files is continued in those folders. The postgresql folder contains the definition of the service and stateful set for the database Concourse uses. The service gets a ClusterIP, because it is only accessed from inside the cluster. The stateful set contains one interesting part: the definition for a persistent volume.
+
+```yaml
+# postgresql/statefulset.yaml:31
+resources:
+  volumeClaimTemplates:
+  - metadata:
+      name: data
+      namespace: concourse
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 5Gi
+```
+
+The secrets template (as all other templates in this repository) contains a list of the needed secrets to run this projct.
+
+In the concourse folder, the definition for the actual Concourse deployment can be found. There are services and stateful sets for both the web page and the workers. There is one web pod and there are three worker pods. The web service gets a NodePort to enable access through the Nginx reverse proxy, the workers with their ClusterIP are only accessable from inside the cluster.
+
+One detail worth discussing is hiding in the definition of the stateful set for the worker. The workers are privileged to allow them to start new containers in their pod. This is needed beause Concourse runs tests in containers and would not be allowed to if not privileged.
```

Comment:
```suggestion
One detail worth discussing is in the definition of the stateful set for the worker. 
The workers are privileged to allow them access to the Docker process of the host system.
They need this access to start new containers in their pod, since Concourse runs tasks in containers and would not be allowed to if not privileged.
```